docs/proposals/network-policy.md

```diff
@@ -0,0 +1,127 @@
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# NetworkPolicy
+
+## Abstract
+
+A proposal for implementing a new resource - NetworkPolicy - which
+will enable definition of ingress policies for selections of pods.
+
+The design for this proposal has been created by, and discussed
+extensively within the Kubernetes networking SIG.  It has been implemented
+and tested using Kubernetes API extensions by various networking solutions already.
+
+In this design, users can create various NetworkPolicy objects which select groups of pods and
+define how those pods should be allowed to communicate with each other.  The
+implementation of that policy at the network layer is left up to the
+chosen networking solution.
+
+> Note that this proposal does not yet include egress / cidr-based policy, which is still actively undergoing discussion in the SIG.  In general,
+egress is expected to augment this proposal rather than modify it, with one exception (noted below).
+
+## Implementation
+
+The implmentation in Kubernetes consists of:
+- A v1beta NetworkPolicy API object
+- A field on the `Namespace` object to control isolation.
+
+### Namespace changes
+
+> Note that this section is subject to some change with the addition of egress policy as mentioned above.
+
+The `Namespace` object will be augmented with the field `networkIsolation` which will take one of the following values:
+
+When `networkIsolation=yes` is set on a namespace:
+- Pods in that Namespace will not be accessible from any other source, unless explicitly allowed by a NetworkPolicy object.
+- Pods in that Namespace will be able to access any other source (egress is uninhibited).
+
+When `networkIsolation=no` is set on a namespace:
+- Pods in that Namespace will be accessible from any other source.
+- Pods in that Namespace will be able to access any other source.
+
+The `networkIsolation` field will default to `false` if not specified.
+
+### NetworkPolicy YAML Definition and Behavior
+
+Once the namespace is isolated, a mechanism to selectively allow traffic into the namespace and between pods within
+the namespace is required.  That is accomplished through ingress rules on `NetworkPolicy`
+objects (of which there can be multiple in a single namespace).  Pods selected by
+a NetworkPolicy should allow any incoming connections that match any
+ingress rule, per the network plugin’s capabilities.
```

Comment:
Just to close the loop here. My initial interpretation of a LabelSelector set exclusion was that it would imply a deny since it is an explicit exclusion of the allowed set. The other interpretation is that the excluded label from the set is simply not included in the allow but the decision remains undecided. Given we go with the latter, I agree that precedence is of no importance for the current proposal.